scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -198,6 +198,18 @@ def test_shortest_path_min_only_random(n):
             p = pred[p]
 
 
+def test_dijkstra_random():
+    n = 10
+    # this random seed was carefully chosen in a search
+    # for reproducers of gh-17782
+    rng = np.random.default_rng(3443)
```

Comment:
Maybe I'm paranoid, but I don't fully trust this to not change on us a few years down the line. Let's actually hard-code it:

```python
n = 10
indices = [0, 4, 4, 5, 7, 9, 0, 6, 2, 3, 7, 9, 1, 2, 9, 2, 5, 6]
indptr = [0, 0, 2, 5, 6, 7, 8, 12, 15, 18, 18]
data = [0.33629, 0.40458, 0.47493, 0.42757, 0.11497, 0.91653, 0.69084, 0.64979, 0.62555, 0.743  , 0.01724, 0.99945, 0.31095, 0.15557, 0.02439, 0.65814, 0.23478, 0.24072]
graph = scipy.sparse.csr_matrix((data, indices, indptr), shape=(n, n))
dijkstra(graph, directed=True, return_predecessors=True)
```

(some line-wrapping needed, but that's the idea)